File: Array/122.Best Time to Buy and Sell Stock II/solution.py
#可以用stack解，但是明显比较慢。
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        stack=[]
        profit=0
        if len(prices)<2:
            return 0
        stack.append(prices[0])
        for i in range(1,len(prices)):
            if prices[i]<stack[-1]:
                currMax=stack[-1]
                currMin=stack[-1]
                while stack:
                    currMin=stack.pop()
                stack.append(prices[i])
                profit=profit+(currMax-currMin)
            elif prices[i]>stack[-1]:
                stack.append(prices[i])
        if stack:
            currMax=stack[-1]
            currMin=stack[-1]
            while stack:
                currMin=stack.pop()
            profit=profit+(currMax-currMin)
        return profit

File: Array/122.Best Time to Buy and Sell Stock II/solution.java
//Design an algorithm to find the maximum profit. You may complete as many transactions as you like (i.e., buy one and sell one share of the stock multiple times).
//只要后面比前面大的都行。这是最简单的方法，stack可解，用python解了一遍。
// Input: [7,1,5,3,6,4]
// Output: 7
// Explanation: Buy on day 2 (price = 1) and sell on day 3 (price = 5), profit = 5-1 = 4.
//              Then buy on day 4 (price = 3) and sell on day 5 (price = 6), profit = 6-3 = 3.
// class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit=0;
        for(int i=1; i<prices.length;i++){
            if (prices[i-1]<prices[i]){
                maxProfit+=(prices[i]-prices[i-1]);
            }
        }
        return maxProfit;
    }
}